Feature/IOPS 1569 Profiling England MessageHeader

KazeemHamzat opened the pull request:
Profiling England MessageHeaader and related assets

KazeemHamzat committed:
Create England-Provenance.xml

Profiling England-Provenence

KazeemHamzat committed:
Resetting Target Profile Reference to UKCore

Resetting Target Profile Reference to UKCore

KazeemHamzat committed:
Update: Extension & Example

Adding Extension & Example

KazeemHamzat committed:
Creating-England-MessageHeader

Creating-England-MessageHeader and related assets.

KazeemHamzat committed:
Fixing CodeSystems

Updating ivalueset dentifier within codesystems

KazeemHamzat committed:
Fixing CodeSystem Issues

Fixing CodeSystem Issues

KazeemHamzat committed:
Fixing CodeSystem & ValueSet Issue

Valueset.name & CodeSystem.rank

digitalhealthmartin commented on StructureDefinition/Extension-England-MessageHeaderIDRreplacement.xml:
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="utf-8"?>
+<StructureDefinition xmlns="http://hl7.org/fhir">
+  <id value="Extension-England-MessageHeaderRreplacement" />
```
Also missing 'ID' - not sure whether ID should be included or not but should be consistent with file name

digitalhealthmartin commented on codesystems/CodeSystem-England-MessageEventEPS.xml:
```diff
@@ -0,0 +1,113 @@
+    </identifier>
+    <version value="1.0.0"/>
+    <name value="EnglandMessageEventEPS"/>
+    <title value="EnglandMessageEvent EPS"/>
```
Title: England Message Event EPS

digitalhealthmartin commented on codesystems/CodeSystem-England-MessageReasonPrescription.xml:
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Remove blank line

digitalhealthmartin commented on valusets/Valuesets-England-MessageEvents.xml:
File name prefix should ValueSet- and not ValueSet**s**-

digitalhealthmartin commented on valusets/Valuesets-England-MessageEvents.xml:
```diff
@@ -0,0 +1,33 @@
+<ValueSet xmlns="http://hl7.org/fhir">
+    <id value="England-MessageEvents" />
+    <url value="https://fhir.nhs.uk/ValueSet/England-MessageEvents" />
+    <version value="1.0.12" />
```
Version value should be 1.0.0

digitalhealthmartin commented on valusets/Valuesets-England-MessageEvents.xml:
```diff
@@ -0,0 +1,33 @@
+    <version value="1.0.12" />
+    <name value="EnglandMessageEvents" />
+    <title value="England Message Events" />
+    <status value="active" />
```
Status should be draft

digitalhealthmartin commented on valusets/Valuesets-England-MessageEvents.xml:
```diff
@@ -0,0 +1,33 @@
+    <date value="2023-06-01" />
+    <publisher value="NHS England" />
+    <contact>
+        <name value="Interoperability Team" />
```
Contact name should be NHS England

digitalhealthmartin commented on valusets/Valuesets-England-MessageReasonCodes.xml:
File name prefix should ValueSet- and not ValueSets-

digitalhealthmartin commented on valusets/Valuesets-England-MessageReasonCodes.xml:
```diff
@@ -0,0 +1,118 @@
+  <url value="https://fhir.nhs.uk/ValueSet/England-MessageReasonCode" />
+  <version value="1.0.0" />
+  <name value="EnglandMessageReasonCode" />
+  <title value="England Message ReasonCode" />
```
Title: ReasonCode -> Reason Code

digitalhealthmartin commented on valusets/Valuesets-England-MessageReasonCodes.xml:
```diff
@@ -0,0 +1,118 @@
+  <version value="1.0.0" />
+  <name value="EnglandMessageReasonCode" />
+  <title value="England Message ReasonCode" />
+  <status value="active" />
```
Status should be draft

digitalhealthmartin commented on valusets/Valuesets-England-MessageReasonCodes.xml:
```diff
@@ -0,0 +1,118 @@
+  <name value="EnglandMessageReasonCode" />
+  <title value="England Message ReasonCode" />
+  <status value="active" />
+  <date value="2023-06-03T12:07:34.8813472+00:00" />
```
Date should be 2023-06-01 - although will be changed again in future anyway

digitalhealthmartin commented on valusets/Valuesets-England-MessageReasonCodes.xml:
```diff
@@ -0,0 +1,118 @@
+  <compose>
+    <include>
+      <system value="https://fhir.nhs.uk/CodeSystem/England-MessageReasonPrescription" />
+      <version value="1.0.1" />
```
Version should be 1.0.0

digitalhealthmartin commented on valusets/Valuesets-England-MessageReasonCodes.xml:
```diff
@@ -0,0 +1,118 @@
+    </parameter>
+    <parameter>
+      <name value="version" />
+      <valueUri value="https://fhir.nhs.uk/CodeSystem/England-MessageReasonPrescription|1.0.1" />
```
Version should be |1.0.0 and not |1.0.1

digitalhealthmartin commented on valusets/Valuesets-England-MessageReasonCodes.xml:
```diff
@@ -0,0 +1,118 @@
+    </parameter>
+    <contains>
+      <system value="https://fhir.nhs.uk/CodeSystem/England-MessageReasonPrescription" />
+      <version value="1.0.1" />
```
Should be 1.0.0

digitalhealthmartin commented on valusets/Valuesets-England-MessageReasonCodes.xml:
```diff
@@ -0,0 +1,118 @@
+    </contains>
+    <contains>
+      <system value="https://fhir.nhs.uk/CodeSystem/England-MessageReasonPrescription" />
+      <version value="1.0.1" />
```
Should be 1.0.0

digitalhealthmartin commented on valusets/Valuesets-England-MessageReasonCodes.xml:
```diff
@@ -0,0 +1,118 @@
+    </contains>
+    <contains>
+      <system value="https://fhir.nhs.uk/CodeSystem/England-MessageReasonPrescription" />
+      <version value="1.0.1" />
```
Should be 1.0.0

digitalhealthmartin reviewed (changes requested):
Some changes required

KazeemHamzat commented on StructureDefinition/England-OperationOutcome.xml:
```diff
@@ -78,7 +78,7 @@
       <path value="OperationOutcome.issue" />
       <constraint>
-        <key value="nhsd-errrorcode" />
+        <key value="nhse-errrorcode" />
```
Noted.

KazeemHamzat committed:
Fixing the issues raised on PR

Fixing the issues raised on PR

APageNHS commented on StructureDefinition/England-MessageHeader.xml:
```diff
@@ -0,0 +1,261 @@
+      <rank value="1" />
+    </telecom>
+  </contact>
+  <description value="This profile defines the NHS England constraints and extensions on the FHIR UKCore resource [UKCore MessageHeader](https://simplifier.net/hl7fhirukcorer4/ukcore-messageheader)" />
```
UKCore should be UK Core

APageNHS commented on StructureDefinition/England-MessageHeader.xml:
```diff
@@ -0,0 +1,261 @@
+      <min value="1" />
+      <type>
+        <code value="Reference" />
+        <targetProfile value="https://fhir.hl7.org.uk/StructureDefinition/UKCore-PractitionerRole" />
```
As per discussion, should not include references

APageNHS commented on StructureDefinition/England-MessageHeader.xml:
```diff
@@ -0,0 +1,261 @@
+      <path value="MessageHeader.sender" />
+      <definition value="The current requirement in NHS Digital API's is to rely on the [Access Tokens and Audit (JWT)](https://developer.nhs.uk/apis/spine-core/security_jwt.html) for sender details.&#xD;&#xA;&#xD;&#xA;For use outside of NHS Digital it is recommended this is populated with either the Organisation ODS code or the Practitioner's professional code." />
+      <type>
+        <code value="Reference" />
```
As per discussion, should not include references

APageNHS commented on StructureDefinition/England-MessageHeader.xml:
```diff
@@ -0,0 +1,261 @@
+      <path value="MessageHeader.responsible" />
+      <type>
+        <code value="Reference" />
+        <targetProfile value="https://fhir.hl7.org.uk/StructureDefinition/UKCore-Practitioner" />
```
As per discussion, should not include references

APageNHS commented on StructureDefinition/England-OperationOutcome.xml:
```diff
@@ -17,7 +17,7 @@
     </telecom>
   </contact>
-  <description value="This profile defines the NHS England constraints and extensions on the UKCore FHIR resource [UKCore OperationOutcome](https://simplifier.net/hl7fhirukcorer4/ukcoreoperationoutcome)" />
+  <description value="This profile defines the NHS England constraints and extensions on the FHIR UKCore resource [UKCore OperationOutcome](https://simplifier.net/hl7fhirukcorer4/ukcoreoperationoutcome)" />
```
should be UK Core

APageNHS commented on codesystems/CodeSystem-England-MessageEventBARS.xml:
```diff
@@ -0,0 +1,47 @@
+<CodeSystem xmlns="http://hl7.org/fhir">
+    <id value="England-MessageEventBARS" />
+    <url value="https://fhir.nhs.uk/CodeSystem/England-MessageEventBARS" />
+    <identifier>
```
should this be included? We don't include them in UK Core

KazeemHamzat replied:
This is common in NHSE. I will document it and raise at second level review.

KazeemHamzat committed:
Fixing PR Issues: England-Provenance

Fixing PR Issues: England-Provenance

KazeemHamzat committed:
Fixing PR Issues: England-MessageHeader

Remove Target profiles, update profile descriptions and removing OperationOutcomeCodes

KazeemHamzat committed:
Fixing PR Issues: England-MessageHeader

Removing England-OperationOutcome-Examples

KazeemHamzat committed:
Fixing PR Issues and removing structure map

Fixing PR Issues and removing structure map

KazeemHamzat committed:
Update Extension-England-ProvenanceAgent.xml

Removing Target Profiles

KazeemHamzat committed:
Removing files not related England-Provenance

Removing files not related England-Provenance

KazeemHamzat committed:
Removed files not related to England-MessageHeader

Removed files not related to England-MessageHeader

digitalhealthmartin commented on StructureDefinition/Extension-England-MessageHeaderRreplacement.xml:
Replacement is spelt wrong

digitalhealthmartin commented on valusets/Valueset-England-MessageEvents.xml:
Valueset -> ValueSet

digitalhealthmartin replied:
Still to be resolved

digitalhealthmartin commented on valusets/Valueset-England-MessageReasonCodes.xml:
Valueset -> ValueSet

digitalhealthmartin replied:
Filename: Valueset -> ValueSet

digitalhealthmartin commented on valusets/Valueset-England-MessageReasonCodes.xml:
```diff
@@ -0,0 +1,118 @@
+    </parameter>
+    <parameter>
+      <name value="version" />
+      <valueUri value="https://fhir.nhs.uk/CodeSystem/England-MessageReasonPrescription|1.0.1" />
```
Version still needs changing from 1.0.1 -> 1.0.0

digitalhealthmartin commented on valusets/Valueset-England-MessageReasonCodes.xml:
```diff
@@ -0,0 +1,118 @@
+    </parameter>
+    <contains>
+      <system value="https://fhir.nhs.uk/CodeSystem/England-MessageReasonPrescription" />
+      <version value="1.0.1" />
```
Version still needs changing from 1.0.1 -> 1.0.0

digitalhealthmartin reviewed (changes requested):
Few minor fixes to be made

KazeemHamzat committed:
Fixing PR Issues: England-MessageHeader

Fixing PR Issues: England-MessageHeader

digitalhealthmartin commented on valusets/Valueset-England-MessageEvents.xml:
Filename: Valueset -> ValueSet

digitalhealthmartin reviewed (changes requested):
Filenames to be renamed

KazeemHamzat committed:
Rename Valueset-England-MessageEvents.xml to ValueSet-England-MessageEvents.xml

Rename Valueset-England-MessageEvents.xml to ValueSet-England-MessageEvents.xml

KazeemHamzat committed:
Fixing PR Issues: England-MessageHeader

Rename Valueset-England-MessageReasonCodes.xml to ValueSet-England-MessageReasonCode.xml

digitalhealthmartin reviewed (dismissed):
Fixes done. Approved

KazeemHamzat committed:
Fixing PR Issues: Update Extension-England-ProvenanceAgent.xml

Update Extension-England-ProvenanceAgent.xml

SBajanki1 commented on StructureDefinition/Extension-England-MessageHeaderLocalPart.xml:
```diff
@@ -17,8 +17,7 @@
   </contact>
-  <description value="Other resources *from the patient record* that may be relevant to the event. The information from these resources was either used to create the instance or is provided to help with its interpretation. This extension **should not** be used if more specific inline elements or extensions are available. For example, use `Observation.hasMember` instead of supportingInformation fo ...
-  <purpose value="The prescription whose release was requested, including the MedicationRequest, Patient, PractitionerRole and Organization details. The information from this resource was either used to create the instance or is provided to help with its interpretation." />
+  <description value="An extension to indicate a specific receiver of a message when routing to a specific application is not sufficient." />
```
purpose is missing

SBajanki1 commented on StructureDefinition/Extension-England-MessageHeaderLocalPart.xml:
```diff
@@ -30,27 +29,37 @@
     <element id="Extension.url">
       <path value="Extension.url" />
-      <fixedUri value="https://fhir.nhs.uk/StructureDefinition/Extension-England-SpineSupportingInfoPrescription" />
+      <fixedUri value="https://fhir.nhs.uk/StructureDefinition/Extension-MessageHeader-LocalPart" />
```
update URI

SBajanki1 commented on StructureDefinition/Extension-England-MessageHeaderReplacement.xml:
```diff
@@ -0,0 +1,55 @@
+      <rank value="1" />
+    </telecom>
+  </contact>
+  <description value="An extension used to track history of message headers (Previous Bundle.identifier)." />
```
purpose is missing

KazeemHamzat committed:
Fixing PR Issues: England-MessageHeader

Fixing PR Issues: England-MessageHeader

SBajanki1 commented on valusets/ValueSet-England-MessageReasonCode.xml:
```diff
@@ -0,0 +1,118 @@
+  <name value="EnglandMessageReasonCode" />
+  <title value="England Message Reason Code" />
+  <status value="draft" />
+  <date value="2023-06-01" />
```
telecom and publisher are missing

SBajanki1 reviewed (changes requested):
Added comments

KazeemHamzat committed:
Fixing PR Issues: England-MessageHeader

Adding CS and fixing PR Issues.

KazeemHamzat committed:
Adding CS: England-MessageHeader

Adding CS: England-MessageHeader

KazeemHamzat committed:
Adding CS: England-Provenance

Adding CS: England-Provenance

KazeemHamzat committed:
Merge branch 'DSTU2_develop' of https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide into Feature-IOPS-1513-Profiling-England-Provenance

KazeemHamzat committed:
Added CS: England-Provenance

Added CS: England-Provenance

KazeemHamzat committed:
Update Extension-England-ProvenanceAgent.xml

Update Extension-England-ProvenanceAgent.xml

KazeemHamzat committed:
Added CS: England-MessageHeader

KazeemHamzat committed:
Merge branch 'DSTU2_develop' of https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide into Feature/IOPS-1569-Profiling-England-MessageHeaader

KazeemHamzat committed:
Reverse Added CS

KazeemHamzat committed:
Update: England-MessageHeader

Update: England-MessageHeader

KazeemHamzat committed:
Update: England-CapabilityStatement.xml

Fixing type for PractitionerRole

SBajanki1 committed:
Merge branch 'DSTU2_develop' into Feature/IOPS-1569-Profiling-England-MessageHeaader
